### DISCUSSION TITLE
Type hinting a class decorator that returns a subclass

### DISCUSSION THREAD

**Discussion by BojieSheng** [python/typing#1529] [Q&A]
Anyone knows who to annotate "cls" and return type for below function "examle_a"?

```
def example_a(cls):
    class A(cls):
        def __init__(self, x, y):
            self.x=x
            self.y=y
            super().__init__(x,y)

    return A
```

cls can be any class.

**Comment by erictraut** [comment]
What you're doing here is too dynamic for static typing — at least for today's static type system. There has been some discussion of adding intersection types to the type system, and that could be used to approximate the type of the value returned by this function, but even that would be only an approximation.

In general, this function can't be made type safe because it's overriding attributes `x` and `y` for the class that's passed into the function, and that class could be anything. If that class happens to already have attributes named `x` and `y`, this function will create a class that fails to work correctly. You might want to look for alternative approaches that don't suffer from this potential fragility.
